jerry-core/parser/js/js-scanner-util.c

```diff
@@ -617,12 +626,14 @@ scanner_pop_literal_pool (parser_context_t *context_p, /**< context */
 #if ENABLED (JERRY_ES2015)
       extended_type |= SCANNER_LITERAL_IS_USED;
 
+      const uint8_t mask = (SCANNER_LITERAL_IS_ARG | SCANNER_LITERAL_IS_LOCAL);
+
       if ((type & SCANNER_LITERAL_IS_ARG)
-          || (!(literal_location_p->type & SCANNER_LITERAL_IS_ARG)
-              && (literal_location_p->type & SCANNER_LITERAL_IS_LOCAL)))
+          || (literal_location_p->type & mask) == SCANNER_LITERAL_IS_LET
```

Comment:
Cannot we use 
` || (literal_location_p->type & mask) == (SCANNER_LITERAL_IS_LET | SCANNER_LITERAL_IS_CONST)` here?

Reply:
Absolutely no. That is a case we want to exclude here, not include. And the two other cases will be excluded, which we want to include :) But I can change the mask to SCANNER_LITERAL_IS_LOCAL, since `(type & SCANNER_LITERAL_IS_ARG)` handles ensures that SCANNER_LITERAL_IS_ARG is not set.

Reply:
I was wrong, those types belong to different literals.
The summary is, we have 3 bits, and from the 8 combinations, we reject these two. I don't think we can simplify it.

Reply:
Sure, thanks for the explanation.